config/src/config/secure_config.rs

```diff
@@ -126,3 +168,79 @@ impl OnDiskStorageConfig {
         self.data_dir = data_dir;
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Write;
+
+    #[test]
+    fn test_token_config_parsing() {
+        #[derive(Debug, Deserialize, PartialEq, Serialize)]
+        struct Config {
+            vault: VaultConfig,
+        }
+
+        let from_config = Config {
+            vault: VaultConfig {
+                namespace: None,
+                server: "127.0.0.1:8200".to_string(),
+                token: Token::FromConfig(TokenFromConfig {
+                    token: "test".to_string(),
+                }),
+            },
+        };
+
+        let text_from_config = "
+[vault]
```

Comment:
nit: is it common practice to do this kind of indentation for multi line strings? Why not match the indentation of the let statement?

Reply:
You know, I don't know, I figured cargo xfmt would fix this if it wasn't right.